packages/@lwc/template-compiler/src/shared/ir.ts

```diff
@@ -42,105 +141,293 @@ export function createElement(original: parse5.Element): IRElement {
         throw new Error('Invalid element AST node. Missing source code location.');
     }
 
+    return location;
+}
+
+export function parseSourceLocation(location: parse5.Location): SourceLocation {
     return {
-        type: 'element',
-        tag: original.tagName,
-        namespace: original.namespaceURI,
+        startLine: location.startLine,
+        startColumn: location.startCol,
+        endLine: location.endLine,
+        endColumn: location.endCol,
+        start: location.startOffset,
+        end: location.endOffset,
+    };
+}
+
+export function literal(value: string | boolean): Literal {
+    return {
+        type: LWCNodeType.Literal,
+        value,
+    };
+}
```

Comment:
You can make this helper function a little smarter by using generics. This would be helpful for the `domDirective` (see comment below). 

```suggestion
export function literal<T extends string | boolean>(value: T): Literal<T> {
    return {
        type: LWCNodeType.Literal,
        value,
    };
}
```

Reply:
This is awesome!  I'm going to explore using generics as an option more.